src/extras/module-types.js

```diff
@@ -40,7 +40,7 @@ import { resolveUrl } from '../common.js';
         return res.text()
         .then(function (source) {
           source = source.replace(/url\(\s*(?:(["'])((?:\\.|[^\n\\"'])+)\1|((?:\\.|[^\s,"'()\\])+))\s*\)/g, function (match, quotes, relUrl1, relUrl2) {
-            return 'url(' + quotes + resolveUrl(relUrl1 || relUrl2, url) + quotes + ')';
+            return ['url(', quotes, resolveUrl(relUrl1 || relUrl2, url), quotes, ')'].join('');
```

Comment:
Can you please explain the bug that this is resolving? This refactoring looks entirely equivalent to me?

Reply:
Sure, I should have probably added the following to the PR description... 

It's valid CSS to not include quotes in urls like `background-image: url(my/image.png);`. However the code above that tries to resolve urls doesn't seem to cater for this. So if `quotes` is undefined the string concatenation adds `undefined` to the result and the image fails to load. e.g.

```javascript
'url(' + undefined + 'my/image.png' + undefined + ')' // --> "url(undefinedmy/image.pngundefined)"
```

However using `array.join` we can take advantage of the [spec](https://262.ecma-international.org/5.1/#sec-15.4.4.5) to discard unwanted undefined from the result.

> If element0 is undefined or null, let R be the empty String; otherwise, Let R be ToString(element0).

e.g.
```javascript
['url(', undefined, 'my/image.png', undefined, ')'].join('') // --> "url(my/image.png)" 
```
I'm more than happy to refactor if you feel the approach I've taken isn't explicit enough.